applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_process_utilities.cpp

```diff
@@ -0,0 +1,126 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+#include "custom_processes/apply_excavation_process.h"
+#include "custom_processes/apply_k0_procedure_process.h"
+#include "custom_utilities/process_utilities.h"
+#include "testing/testing.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(GetModelPartsFromSettings_SingleModelPart, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    model.CreateModelPart("Main");
+
+    Parameters settings(R"(
+        {
+            "model_part_name": "Main"
+        })");
+
+    auto model_parts = ProcessUtilities::GetModelPartsFromSettings(model, settings, "TestProcess");
+
+    KRATOS_CHECK_EQUAL(model_parts.size(), 1);
+    KRATOS_CHECK_EQUAL(model_parts[0].get().Name(), "Main");
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetModelPartsFromSettings_ListOfModelParts, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    model.CreateModelPart("Part1");
+    model.CreateModelPart("Part2");
+
+    Parameters settings(R"(
+        {
+            "model_part_name_list": ["Part1", "Part2"]
+        })");
+
+    auto model_parts = ProcessUtilities::GetModelPartsFromSettings(model, settings, "TestProcess");
+
+    KRATOS_CHECK_EQUAL(model_parts.size(), 2);
+    KRATOS_CHECK_EQUAL(model_parts[0].get().Name(), "Part1");
+    KRATOS_CHECK_EQUAL(model_parts[1].get().Name(), "Part2");
+}
+
+struct NamedProcessFactory {
+    std::string                                                                         name;
+    std::function<std::unique_ptr<Kratos::Process>(Kratos::Model&, Kratos::Parameters)> factory;
```

Comment:
Shouldn't the second type be `const Kratos::Parameters&`? The lambda expressions that you supply later pass by reference-to-const, not by value. Or do I overlook something?
```suggestion
    std::function<std::unique_ptr<Kratos::Process>(Kratos::Model&, const Kratos::Parameters&)> factory;
```

Reply:
You are right and it works.